apps/designer/app/designer/features/style-panel/sections/layout/layout.tsx

```diff
@@ -146,53 +130,72 @@ const LayoutSectionFlex = ({
   );
 };
 
+const orderDisplayValuesFirst = [
+  "block",
+  "flex",
+  "inline-block",
+  "inline-flex",
+  "inline",
+  "none",
+];
+
+const compareDisplayValues = (a: { name: string }, b: { name: string }) => {
+  const aIndex = orderDisplayValuesFirst.indexOf(a.name);
+  const bIndex = orderDisplayValuesFirst.indexOf(b.name);
+  return aIndex - bIndex;
+};
+
 export const LayoutSection = ({
-  setProperty,
   deleteProperty,
   createBatchUpdate,
   currentStyle,
   sectionStyle,
-  inheritedStyle,
-  category,
   styleConfigsByCategory,
-  moreStyleConfigsByCategory,
 }: RenderCategoryProps) => {
-  const ActiveLayout = layouts.get(toValue(currentStyle.display));
+  const displayValue = toValue(currentStyle.display);
 
-  if (ActiveLayout) {
-    return (
-      <ActiveLayout
-        setProperty={setProperty}
-        deleteProperty={deleteProperty}
-        createBatchUpdate={createBatchUpdate}
-        currentStyle={currentStyle}
-        sectionStyle={sectionStyle}
-        inheritedStyle={inheritedStyle}
-        category={category}
-        styleConfigsByCategory={styleConfigsByCategory}
-        moreStyleConfigsByCategory={moreStyleConfigsByCategory}
-      />
-    );
-  }
+  const { display } = sectionStyle;
 
   return (
     <>
-      <ShowMore
-        styleConfigs={moreStyleConfigsByCategory.map((entry) =>
-          renderProperty(entry)
-        )}
-      />
-      {styleConfigsByCategory.map((entry) => renderProperty(entry))}
+      {display?.styleConfig && (
+        <Grid css={{ gridTemplateColumns: "4fr 6fr" }}>
+          <PropertyName
+            property="display"
+            label={display.styleConfig.label}
+            onReset={() => deleteProperty("display")}
+          />
+          <SelectControl
+            category={display.category}
+            currentStyle={display.currentStyle}
+            inheritedStyle={display.inheritedStyle}
+            setProperty={display.setProperty}
+            // show only important values first and hide others with scroll
+            styleConfig={{
+              ...display.styleConfig,
+              items: display.styleConfig.items
```

Comment:
why do you need to use `display.styleConfig.items` at all if the list is fixed and already sorted?

Reply:
Labels